racun-service/src/main/java/rs/edu/raf/banka/racun/service/impl/UgovorService.java

```diff
@@ -57,44 +63,122 @@ public TransakcionaStavka getTransakcionaStavkaById(Long id)
         return ugovor.get();
     }
 
-    public List<Ugovor> getAll()
+    private UserDto getUserByToken(String token) throws Exception {
+        var user = userService.getUserByToken(token);
+        if(user == null)
+            throw new Exception("Invalid token");
+        return user;
+    }
+
+    private boolean isUserSupervisor(UserDto user)
+    {
+        return user.getRoleName().equals("ROLE_SUPERVISOR")
+                || user.getRoleName().equals("ROLE_ADMIN");
+    }
+
+    private boolean isUserAgent(UserDto user)
     {
-        return ugovorRepository.findAll();
+        return user.getRoleName().equals("ROLE_AGENT");
+    }
+
+    private void checkPermissions(Ugovor ugovor, String token) throws Exception {
+
+        var user = getUserByToken(token);
+
+        if(isUserSupervisor(user))
+            return;
+        if(isUserAgent(user) && ugovor.getAgentId() == user.getId())
+            return;
+
+        throw new Exception("No permissions");
+    }
+
+    private void checkPermissions(String token) throws Exception {
```

Comment:
Preimenuj ove funckije da detaljnije opisiuju šta tačno rade. `checkPermissions` je previše generično, pogotovo što se dva puta javlja.